projects/plugins/jetpack/modules/sharedaddy/sharing-service.php

```diff
@@ -687,7 +689,9 @@ public function __construct( $id, $total ) {
 		$this->service = $services->get_service( $id );
 		$this->total   = (int) $total;
 
-		$this->name = $this->service->get_name();
+		if ( is_object( $this->service ) ) {
```

Comment:
Would it work to be more specific, and check for the `Sharing_Source` instead, so we know `get_name` will be in there?

Reply:
Yes, good point. I was also able to come up with steps to replicate the warning and test the fix now for this specific change too (although it looks straightforward to check if it's an instance of `Sharing _Source`, I wanted to be sure there wouldn't be additional errors) , so updated the testing instructions.